lib/src/visualisation.dart

```diff
@@ -0,0 +1,156 @@
+part of many_points;
+
+class Visualisation {
+  String _imagePrefix = 'output/image/';
+
+  List<DataTransformFunction> _dataTransforms = [];
+  ColorTransformFunction _colorTransform;
+  List<int> xs = [];
+  List<int> ys = [];
+  List<num> values = [];
+  List<int> colors = [];
+  Range xRange;
+  Range yRange;
+  Range valueRange;
+
+  Visualisation() {
+    xRange = new Range(double.INFINITY, double.NEGATIVE_INFINITY);
+    yRange = new Range(double.INFINITY, double.NEGATIVE_INFINITY);
+    valueRange = new Range(double.INFINITY, double.NEGATIVE_INFINITY);
+  }
+
+  /// Add a data point defined by the [x] and [y] coordinates and its [value].
+  void addData(num x, num y, num value) {
+    xs.add(x);
+    ys.add(y);
+    values.add(value);
+
+    // Adjust the ranges for the data.
+    xRange.min = math.min(xRange.min, x);
+    xRange.max = math.max(xRange.max, x);
+
+    yRange.min = math.min(yRange.min, y);
+    yRange.max = math.max(yRange.max, y);
+
+    valueRange.min = math.min(valueRange.min, value);
+    valueRange.max = math.max(valueRange.max, value);
+  }
+
+  /// Appends the data points defined by the coordinate lists [ix] and [iy] and
+  /// the value list [ivalue].
+  void addAllDataPoints(
```

Comment:
There are some potentially nasty bugs here. If someone messes up and passes 10 xs, 5 ys, and 10 values and then passes another 10, x y and values, you'll end up with data corruption in the internal data structures with some y values to different xs. I'd rather that never happened.

I think the safest way is to operate on the iterators manually calling next on them to get values and then making calls to addValue one tuple at a time.